### ISSUE TITLE
Current working directory is not consistent when developing addons

### ISSUE THREAD

**Issue by aexmachina** [ember-cli/ember-cli#4258]
The root cause for [this issue](https://github.com/aexmachina/ember-cli-sass/issues/56) is that when developing an app `process.cwd()` is the root of the app, but when developing an addon `process.cwd()` is the root of the addon. This may be expected behaviour, but the question is how we should handle both cases.

For example, ember-cli-sass inspects the file system to see whether a `.scss` or a `.sass` file is being used, and assumes that the current working directory is the app. How can I get the root directory of the app in a way that returns `${addon root}/tests/dummy` when run from inside an addon?

**Comment by stefanpenner** [comment]
Maybe a doc problem. But the cwd will always be the root of the primary project. So when developing addons we must take this into account.
 
Another (and likely better approach) is to expand dummy app to behave more like a real app. So the file path issues will become apparent. Eg, move the brocfile to the dummy itself.

**Comment by aexmachina** [comment]
Thanks @stefanpenner, any suggestions how I can get the root directory of the app in a way that returns `${addon root}/tests/dummy` when run from inside an addon?

**Comment by dukex** [comment]
There are good reason for the preprocessors code be too diferentes? 

[models/addon.js](https://github.com/ember-cli/ember-cli/blob/4cff0dbb3163edcaf51fc3b204398f0945448408/lib/models/addon.js#L399-L408)
```
Addon.prototype.compileStyles = function(tree) {
  this._requireBuildPackages();

  if (tree) {
    return preprocessCss(tree, '/', '/', {
      outputPaths: { 'addon': this.name + '.css' },
      registry: this.registry
    });
  }
};
```

[broccoli/ember-app.js](https://github.com/ember-cli/ember-cli/blob/4cff0dbb3163edcaf51fc3b204398f0945448408/lib/broccoli/ember-app.js#L971-L973)
```
  var options = { outputPaths: this.options.outputPaths.app.css };
  options.registry = this.registry;
  var preprocessedStyles = preprocessCss(stylesAndVendor, '/app/styles', '/assets', options);
```

**Comment by dukex** [comment]
@aexmachina the `tree` params received by `toTree` returns the ${root addon}

**Comment by mattmarcum** [comment]
I was trying to develop an addon and ran into this problem - the dummy app wasn't getting any of its styles compiled into dummy.css.  I tried all the different hacks and work a rounds that I could find - adding an `app/styles/app.scss`, trying out the branch above from @dukex - none of these fixed the basic problem.  The best I could do was get an empty dummy.css file to generate and then I could see my actual styles in the dummy.css.map file(like this: `../../tree_merger-tmp_dest_dir-HLO9CoWO.tmp/app/styles/app.scss"`).  But no matter what I can't get my dummy's styles to actually compile into dummy.css.

**Comment by justin-lau** [comment]
@mattmarcum #4435 is preventing the add-on options to be passed correctly. It should be fixed in 1.13.2. Until then you have to stick with 0.2.7 to use ember-cli-sass.
